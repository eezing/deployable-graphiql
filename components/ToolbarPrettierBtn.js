import React from 'react';
import PropTypes from 'prop-types';
import GraphiQL from 'graphiql';
import prettier from 'prettier/standalone';
import prettierGraphql from 'prettier/parser-graphql';

const ToolbarHistoryBtn = ({ graphiqlRef }) => {
  function _onClick() {
    const editor = graphiqlRef.current.getQueryEditor();
    editor.setValue(
      prettier.format(editor.getValue(), {
        parser: 'graphql',
        plugins: [prettierGraphql]
      })
    );
  }

  return (
    <GraphiQL.Button onClick={_onClick} label="Prettier" title="Prettier" />
  );
};

ToolbarHistoryBtn.propTypes = {
  graphiqlRef: PropTypes.object.isRequired
};

export default ToolbarHistoryBtn;
