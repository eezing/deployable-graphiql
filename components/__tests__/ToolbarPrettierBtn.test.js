import React from 'react';
import renderer from 'react-test-renderer';
import ToolbarPrettierBtn from '../ToolbarPrettierBtn';

test('ToolbarPrettierBtn snapshot', () => {
  const dummyRef = React.createRef();
  const component = renderer.create(
    <ToolbarPrettierBtn graphiqlRef={dummyRef} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
