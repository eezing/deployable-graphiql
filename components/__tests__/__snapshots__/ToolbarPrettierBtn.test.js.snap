// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarPrettierBtn snapshot 1`] = `
<a
  className="toolbar-button"
  onClick={[Function]}
  onMouseDown={[Function]}
  title="Prettier"
>
  Prettier
</a>
`;
